hazelcast/src/main/resources/hazelcast-config-5.1.json

```diff
@@ -1494,10 +1494,26 @@
                 "additionalProperties": false,
                 "properties": {
                   "capacity": {
-                    "type": "string",
-                    "pattern": "^[1-9]([0-9]*) (B|KB|MB|GB)$",
-                    "default": "256 MB",
-                    "description": "Memory tier capacity, i.e., how much main memory should this tier consume at most. The capacity should follow the pattern [1-9]([0-9]*) (B|KB|MB|GB). For instance, capacity=\"1024 MB\" is valid, whereas capacity=\"2^10 mb\" is not."
+                    "type": "object",
+                    "description": "Memory tier capacity, i.e., how much main memory should this tier consume at most. The default is 256 megabytes.",
+                    "properties": {
+                      "unit": {
+                        "enum": [
+                          "BYTES",
+                          "KILOBYTES",
+                          "MEGABYTES",
+                          "GIGABYTES"
+                        ],
+                        "default": "MEGABYTES",
+                        "description": "Unit can be bytes, kilobytes, megabytes and gigabytes. Default unit is MEGABYTES."
+                      },
+                      "value": {
+                        "type": "number",
+                        "default": 256,
+                        "minimum": 1,
+                        "description": "Default value is 256."
+                      }
+                    }
```

Comment:
Can't we introduce a type and replace `capacity` here and in `native-memory`? We'll need it in the disk tier config as well later. It seems `SyncBackups` and the likes can be used as examples.

Reply:
Nice point, thanks :) Addressed in https://github.com/hazelcast/hazelcast/pull/20476/commits/df4e2daeb831aae73a68268b7e8dcc2671f9fddd My initial concern was that native-memory and tiered-store has different default values for memory-sizes. I think that should not be a problem, since even though in xml/yaml config default value for native-memory is 128 MB, in the implementation it is actually 512 MB